tensorflow/tools/compatibility/tf_upgrade_v2.py

```diff
@@ -1177,6 +1177,53 @@ def __init__(self):
         for name, new_name in self.symbol_renames.items()
     }
 
+  @staticmethod
+  def _extract_glimpse_transformer(parent, node, full_name, name, logs, errors):
+    def _replace_uniform_noise_node(parent, old_value):
+      """Replaces old_value with 'uniform' or 'guassian'."""
+      uniform = ast.Str(s="uniform")
+      gaussian = ast.Str(s="gaussian")
+      new_value = ast.IfExp(body=uniform, test=old_value,
+                            orelse=gaussian)
+      # This copies the prefix and suffix on old_value to new_value.
+      pasta.ast_utils.replace_child(parent, old_value, new_value)
+      ast.copy_location(new_value, old_value)
+      # Put parentheses around noise.value (and remove the old prefix/
+      # suffix, they should only be around new_value).
+      pasta.base.formatting.set(new_value, "prefix", "(")
+      pasta.base.formatting.set(new_value, "suffix", ")")
```

Comment:
You need these around new_value.test, not new_value, since you're making something that looks like `"uniform if blah else "gaussian"`, and if `blah` is something like `a if b else c + d`, then the resulting expression is `"uniform" if a if b else c else "gaussian"` which is a syntax error, but `"uniform" if (a if b else c) else "gaussian"` is valid.

Reply:
Done.